nntrainer/utils/profiler.h

```diff
@@ -163,6 +194,22 @@ class GenericProfileListener : public ProfileListener {
   virtual void report(std::ostream &out) const override;
 
 private:
+  /**
+   * @brief Called when time event occurs
+   *
+   */
+  void onNotifyTimeEvent(PROFILE_EVENT event, const int time_item,
+                         const std::string &str,
+                         const std::chrono::microseconds &duration);
+
+  /**
+   * @brief Called when memory event occurs
+   *
+   */
+  void onNotifyMemoryEvent(PROFILE_EVENT event, const size_t total_alloc_size,
```

Comment:
I think it is ok for now to add the memory notify feature in GeneralProfileListener. 
This GenericProfileListener is for time profiling, not memory profiling. It might be better to make MemoryListener by inheriting ProfileListener later.

Reply:
OK. I'll expanding memory profiliing feature in the later.